### DISCUSSION TITLE
How to get the parameters for fast pitch fine-tuning?

### DISCUSSION THREAD

**Discussion by godspirit00** [NVIDIA-NeMo/NeMo#3138] [Q&A] [answered]
Hello,
I see in the fast pitch fine-tuning notebook that some parameters need to be set according to the dataset.
```
model.pitch_avg=214.5 model.pitch_std=30.9 model.pitch_fmin=80 model.pitch_fmax=512
```
So how can I get the values for these parameters if I'm to fine-tune on a custom dataset?

And also, are the ```pitch_avg``` and ```pitch_std``` the same things as ```mean``` and ```std``` in the fast pitch inference pitch control notebook?

Sorry for the dumb questions...
Thanks for your time and work!

**Comment by XuesongYang** [comment, marked as answer]
Hi @godspirit00, we have updated the finetuning FastPitch tutorial here: https://github.com/NVIDIA/NeMo/blob/main/tutorials/tts/FastPitch_Finetuning.ipynb. Could you have a look?

> So how can I get the values for these parameters if I'm to fine-tune on a custom dataset?

I would suggest estimating `pitch_mean`, `pitch_std`, `pitch_min`, and `pitch_max` from your custom dataset. You could re-use the script: https://github.com/NVIDIA/NeMo/blob/main/scripts/dataset_processing/tts/extract_sup_data.py

**Reply by godspirit00** [reply]
Thanks! I'll have a try!
